cli/src/template_builder.rs

```diff
@@ -2137,8 +2137,17 @@ mod tests {
         }
 
         fn parse_err(&self, template: &str) -> String {
-            let err = self.parse(template).err().unwrap();
-            iter::successors(Some(&err), |e| e.origin()).join("\n")
+            let Err(orig_err) = self.parse(template) else {
+                panic!("Got unexpected successful template rendering");
+            };
+
+            let mut err: Option<&dyn std::error::Error> = Some(&orig_err);
+            let mut out = String::new();
+            while let Some(inner) = err {
+                writeln!(&mut out, "{inner}").unwrap();
+                err = std::error::Error::source(inner);
+            }
+            out
```

Comment:
nit: this can be `iter::successors(Some(&err as &dyn std::error::Error), |e| e.source()).join("\n")`. we don't care about the extra allocation cost.